main/entry.c

```diff
@@ -635,6 +641,7 @@ static size_t appendSourceLine ( void putc_func (char , void *), const char *con
 		putc_func (c, data);
 		++length;
 	}
+
 	return length;
```

Comment:
The length value is returned here, so why not use it at top level to decide if `$` should be appended, instead of passing `omitted` all across the hierarchy?